helpers/securewebhook/securewebhook.go

```diff
@@ -27,3 +38,25 @@ func GetRequestBody(s *Settings) ([]byte, error) {
 	}
 	return b, nil
 }
+
+// VerifySignature uses the ECDSA publicKey and, verifies received payload and signature
+func VerifySignature(publicKey *ecdsa.PublicKey, payload []byte, signature, timestamp string) (bool, error) {
+	signatureBytes, err := base64.StdEncoding.DecodeString(signature)
+	if err != nil {
+		return false, err
+	}
+
+	ecdsaSig := &RS{}
+	_, err = asn1.Unmarshal(signatureBytes, ecdsaSig)
+	if err != nil {
+		return false, err
+	}
+
+	h := sha256.New()
+	if timestamp != "" {
```

Comment:
timestamp is not optional and is required as part of the signature

Reply:
This is just a check, how does that matter? Do you want me to remove the check and always add this in the timestamp?

Reply:
This conditional makes it seem that timestamp is optional when it is not.

Reply:
Timestamp is required as an argument, how does this condition make it seem as optional? And this is just the double check if it is empty then do not add in the hash.

Reply:
The user can pass the empty string "" as the timestamp, which would not be a valid value. There is no need to double check if it is not empty since `h.Write("") is the same as not doing h.Write()`. 
https://goplay.space/#rr5y_9R01CW

Reply:
@thavanle should we write an assert or do an argument check to see timestamp is not ""?

Reply:
Good point @anujs3, if we are saying that these input fields are required then we should probably do validation for them.

Simple length check and empty string should be sufficient I think, but we should ask DI on their opinion for these (an example of data validation can be seen https://github.com/sendgrid/sendgrid-go/blob/master/helpers/mail/mail_v3.go#L713)

Reply:
I'm fine with the current implementation.